### PR TITLE
GraphQL error logging improvement

### LINKED ISSUE #2962: Improved GraphQL error logging
We are running Spectrum for the first time and seeing the same issue as #2950. While trying to understand what's going on, we found out that the GraphQL error we are getting is not very informative:
```
  api:utils:error-formatter ---GraphQL Error--- +129ms
  api:utils:error-formatter Error: Cannot return null for non-nullable field Thread.channel.
  api:utils:error-formatter     at completeValue (.../spectrum/node_modules/graphql/execution/execute.js:596:13)
  api:utils:error-formatter     at .../spectrum/node_modules/graphql/execution/execute.js:582:14
  api:utils:error-formatter     at <anonymous>
  api:utils:error-formatter ------------------- +0ms
```
Here is how the same error message looks after we extended error logging locally:
```
  api:utils:error-formatter ---GraphQL Error--- +129ms
  api:utils:error-formatter Error: Cannot return null for non-nullable field Thread.channel.
  api:utils:error-formatter     at completeValue (.../spectrum/node_modules/graphql/execution/execute.js:596:13)
  api:utils:error-formatter     at .../spectrum/node_modules/graphql/execution/execute.js:582:14
  api:utils:error-formatter     at <anonymous> +0ms
  api:utils:error-formatter ---GraphQL Query--- +0ms
  api:utils:error-formatter query getCommunityThreadConnection($id: ID, $after: String)  +0ms
  api:utils:error-formatter { id: '1', after: null } +0ms
  api:utils:error-formatter -------------------
```
Let me know if there is any interest in this and I will submit a PR.

### PR DESCRIPTION
**Pull request by a-golovanov**
**Status**
- [ ] WIP
- [x] Ready for review
- [ ] Needs testing

**Deploy after merge (delete what needn't be deployed)**
- api

Added GraphQL query name, variables and path where the error occurred,
if any. See test snapshots for an example.

Closes #2962, cc @brianlovin

### CODE REVIEW

File: api/test/utils/__snapshots__/create-graphql-error-formatter.test.js.snap [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+",
+]
+`;
+
+exports[`createGraphQLErrorFormatter logs query and variables 1`] = `
+Array [
+  "---GraphQL Error---",
+  "GraphQLError: Cannot return null for non-nullable field Thread.channel.",
+  "query getCommunityThreadConnection( $id: ID )",
+  "variables [object Object]",
```
Comment (mxstbr) [inline api/test/utils/__snapshots__/create-graphql-error-formatter.test.js.snap]: We should probably stringify these to make this log useful

**Review by mxstbr — COMMENTED** [review]
Thanks for digging into this, this is much better! One nitpick then let's ship this

**Review by mxstbr — APPROVED** [review]
Looks very great, thank so much! 😍 Will defer to @brianlovin for last signoff

**Review by brianlovin — APPROVED** [review]

**Comment by brianlovin** [comment]
Thank you so much! This will be very useful

**Pull request merged**